Gems/EditorPythonBindings/Code/Source/PythonSystemComponent.cpp

```diff
@@ -258,6 +258,85 @@ namespace EditorPythonBindings
         void Finalize() override {}
     };
 
+    // Manages the acquisition and release of the Python GIL (Global Interpreter Lock).
+    // Used by PythonSystemComponent to lock the GIL when executing python.
+    class PythonSystemComponent::PythonGILScopedLock final
```

Comment:
Expected to see some simple unit tests of this new class, not just a multithreaded integration-with-mocks test (which is also important)

Reply:
This class is an internal helper for `PythonSystemComponent` to lock the Python GIL, to make unit tests on it I'd have to move it to the header and make it publicly accessible, which is something I'd rather avoid. `PythonSystemComponent` does have unit tests, which by extension touch this class's logic.